Assets/MixedRealityToolkit-Examples/UX/Scripts/ProgressExamples.cs

```diff
@@ -0,0 +1,154 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+using System.Collections;
+using MixedRealityToolkit.UX.Progress;
+using MixedRealityToolkit.InputModule;
+using MixedRealityToolkit.InputModule.InputSources;
+using UnityEngine.XR.WSA.Input;
+using MixedRealityToolkit.Utilities.Solvers;
+using UnityEngine.XR.WSA;
+
+namespace MixedRealityToolkit.Examples.UX
+{
+    public class ProgressExamples : MonoBehaviour
+    {
+        public GameObject SceneObject;
+
+        [Header("How long to spend on each stage of loading")]
+        public float LeadInTime = 1.5f;
+        public float LoadingTime = 5f;
+        public float FinishTime = 1.5f;
+
+        [Header("Set these to override the defaults set in the ProgressIndicator prefab")]
+        public GameObject LoadingPrefab = null;
+        public Texture2D LoadingIcon = null;
+
+        [Header("Messages displayed during loading")]
+        public string LeadInMessage = "Lead in Message";
+        public string LoadTextMessage = "Loading with message only";
+        public string LoadOrbsMessage = "Loading with Orbs";
+        public string LoadIconMessage = "Loading with Icon";
+        public string LoadPrefabMessage = "Loading with Prefab";
+        public string LoadProgressMessage = "Loading with Progress";
+        public string LoadProgressBarMessage = "Loading with Bar";
+        public string FinishMessage = "Finished!";
+
+        private void Start()
+        {
+            // Optimize the content for immersive headset
+            if (HolographicSettings.IsDisplayOpaque)
+            {
+               GameObject buttonCollection = GameObject.Find("ButtonCollection");
+               buttonCollection.transform.localScale = new Vector3(1.3f, 1.3f, 1.0f);
+            }
+        }
+
+        public void LaunchProgress(GameObject obj)
+        {
+            Debug.Log("Loading with button " + obj.name);
+
+            if (ProgressIndicator.Instance.IsLoading)
+                return;
+
+            switch (obj.name)
```

Comment:
Not sure if I like that we're switching by the object's name.  What if I want to name my button something else?